src/primitives.cc

```diff
@@ -190,10 +190,16 @@ AbstractNode *PrimitiveModule::instantiate(const Context *ctx, const ModuleInsta
 	case primitive_type_e::CUBE: {
 		auto size = c.lookup_variable("size");
 		auto center = c.lookup_variable("center");
-		size->getDouble(node->x);
-		size->getDouble(node->y);
-		size->getDouble(node->z);
-		size->getVec3(node->x, node->y, node->z);
+		if(size != ValuePtr::undefined){
+			bool converted=false;
+			converted |= size->getDouble(node->x);
+			converted |= size->getDouble(node->y);
+			converted |= size->getDouble(node->z);
+			converted |= size->getVec3(node->x, node->y, node->z);
+			if(!converted){
+				PRINTB("WARNING: Unable to convert CUBE size=%s to a number or a vec3 of numbers, %s", size->toString() % inst->location().toString());
```

Comment:
I wonder if we should print `CUBE` in lower case here, perhaps even `cube(size=%s)`.
Same goes for other similar print statements.

Reply:
`cube` is fine by me. `cube(size=%s)` can get confusing, as we do not show the state of for e.g. `center`. Note that cube is the simplest case, other modules have a much more complicated signature.
We could use [Ellipsis](https://en.wikipedia.org/wiki/Ellipsis) to hint add the omission like so `cube(size=%s...)` or so `cube(size=%s, ...)`. Note that we have also have highly complicated built ins like:
`polyhedron(..., triangles = [[0,1,2,3],[5,4,3,2],[0,4,5,1],[0,3,4],[5,2]], ...);`.

Note that it gets ugly when strings come in to play (`cube("[1,2,3]");`) - but this is an issue either-way.